components/ItemCard.tsx

```diff
@@ -2,31 +2,21 @@ import React from 'react';
 import Image from 'next/image';
 import Link from 'next/link';
 import ItemDetails from './ItemDetails';
+import { PartialItem } from '@/types/supabaseTypes';
 
-type ItemCardPropType = {
-  imageSrc: string;
-  item_name: string;
-  condition: string;
-  item_type: string;
-  postcode: string;
-  postable: boolean;
-  itemId: number;
-  reserved: boolean;
-};
-
-const ItemCard: React.FC<ItemCardPropType> = ({
+const ItemCard: React.FC<PartialItem> = ({
```

Comment:
yeah I think you made a good call here.

I hate having those times ` | undefined` and the prop types should have been stopping it, but I think it arcs back to the schema and isn't as simple a fix as that 🙃